client/components/advanced-filters/index.js

```diff
@@ -0,0 +1,237 @@
+		this.addFilter = this.addFilter.bind( this );
+		this.removeFilter = this.removeFilter.bind( this );
+		this.clearAllFilters = this.clearAllFilters.bind( this );
+	}
+
+	onMatchChange( value ) {
+		this.setState( {
+			match: find( matches, match => value === match.value ),
+		} );
+	}
+
+	onFilterChange( key, property, value ) {
+		const activeFilters = this.state.activeFilters.map( activeFilter => {
+			if ( key === activeFilter.key ) {
+				return Object.assign( {}, activeFilter, { [ property ]: value } );
+			}
+			return activeFilter;
+		} );
+
+		this.setState( { activeFilters } );
+	}
+
+	removeFilter( key ) {
+		const activeFilters = [ ...this.state.activeFilters ];
+		const index = findIndex( activeFilters, filter => filter.key === key );
+		activeFilters.splice( index, 1 );
+		this.setState( { activeFilters } );
+	}
+
+	getTitle() {
+		const { match } = this.state;
+		const { filterTitle } = this.props;
+		return (
+			<Fragment>
+				<span>{ sprintf( __( '%s Match', 'wc-admin' ), filterTitle ) }</span>
+				<SelectControl
+					className="woocommerce-advanced-filters__title-select"
+					options={ matches }
+					value={ match.value }
+					onChange={ this.onMatchChange }
+					aria-label={ __( 'Match any or all filters', 'wc-admin' ) }
+				/>
+				<span>{ __( 'Filters', 'wc-admin' ) }</span>
+			</Fragment>
+		);
+	}
+
+	getSelector( filter ) {
+		const filterConfig = this.props.config[ filter.key ];
+		const { input } = filterConfig;
+		if ( ! input ) {
+			return null;
+		}
+		if ( 'SelectControl' === input.component ) {
+			return (
+				<SelectControl
+					className="woocommerce-advanced-filters__list-select"
+					options={ input.options }
+					value={ filter.value }
+					onChange={ partial( this.onFilterChange, filter.key, 'value' ) }
+					aria-label={ sprintf( __( 'Select %s', 'wc-admin' ), filterConfig.label ) }
+				/>
+			);
+		}
+		if ( 'FormTokenField' === input.component ) {
+			return (
+				<FormTokenField
+					value={ filter.value }
+					onChange={ partial( this.onFilterChange, filter.key, 'value' ) }
+					aria-label={ sprintf( __( 'Add %s', 'wc-admin' ), filterConfig.label ) }
+				/>
+			);
+		}
+		return null;
+	}
+
+	getAvailableFilterKeys() {
+		const { config } = this.props;
+		const activeFilterKeys = this.state.activeFilters.map( f => f.key );
+		return difference( Object.keys( config ), activeFilterKeys );
+	}
+
+	addFilter( key, onClose ) {
+		const filterConfig = this.props.config[ key ];
+		const newFilter = { key, rule: filterConfig.rules[ 0 ] };
+		if ( filterConfig.input && filterConfig.input.options ) {
+			newFilter.value = filterConfig.input.options[ 0 ];
+		}
+		this.setState( state => {
+			return {
+				activeFilters: [ ...state.activeFilters, newFilter ],
+			};
+		} );
+		onClose();
+		// after render, focus the newly added filter's first focusable element
+		setTimeout( () => {
+			const addedFilter = this.filterListRef.current.querySelector( 'li:last-of-type select' );
```

Comment:
Can we focus on the fieldset itself? That might read out better than dropping onto the select input.